core/src/main/scala/sttp/tapir/server/ServerDefaults.scala

```diff
@@ -50,16 +50,17 @@ object ServerDefaults {
       import sttp.tapir.server.{DefaultDecodeFailureResponse => fr}
 
       failingInput(ctx) match {
-        case _: EndpointInput.Query[_]             => Some(fr.status(StatusCode.BadRequest))
-        case _: EndpointInput.QueryParams[_]       => Some(fr.status(StatusCode.BadRequest))
-        case _: EndpointInput.Cookie[_]            => Some(fr.status(StatusCode.BadRequest))
-        case _: EndpointIO.Header[_]               => Some(fr.status(StatusCode.BadRequest))
-        case _: EndpointIO.Headers[_]              => Some(fr.status(StatusCode.BadRequest))
-        case _: EndpointIO.Body[_, _]              => Some(fr.status(StatusCode.BadRequest))
-        case _: EndpointIO.StreamBodyWrapper[_, _] => Some(fr.status(StatusCode.BadRequest))
+        case _: EndpointInput.Query[_]                                             => Some(fr.status(StatusCode.BadRequest))
+        case _: EndpointInput.QueryParams[_]                                       => Some(fr.status(StatusCode.BadRequest))
+        case _: EndpointInput.Cookie[_]                                            => Some(fr.status(StatusCode.BadRequest))
+        case _: EndpointIO.Header[_]                                               => Some(fr.status(StatusCode.BadRequest))
+        case fh: EndpointIO.FixedHeader[_] if fh.h.name == HeaderNames.ContentType => Some(fr.status(StatusCode.UnsupportedMediaType))
```

Comment:
We should also check (dynamic) `Header`, if the decode failure is an `Mismatch` or `Multiple`. If we get other failures, I think we should return 400 like now, both for the dynamic and fixed header case.

Reply:
So you mean to add another cases for every Header Mismatch or Multiple decode failure, but the one with UnsupportedMediaType is ok ?

Reply:
I think it makes sense to return `StatusCode.UnsupportedMediaType` only if there's a mismatch/multiple error. If there's e.g. no value (Missing), either when using a fixed or dynamic header, then we should return a BadRequest. As if there's no value, we can't say that we don't support that media type :)

Reply:
corrected